### PR TITLE
SQL-1867: implement additional connection attributes

### PR DESCRIPTION
**Pull request by jchemburkar**
These connection and column attributes allow for SSMS to relay correct errors with a query (both from the application and driver side). Previously, it would output cached errors caused by these.

### CODE REVIEW

**Review by nbagnard — APPROVED** [review]
LGTM

File: odbc/src/api/functions.rs [hunk 1]
```diff
@@ -1454,6 +1456,14 @@ unsafe fn sql_get_connect_attrw_helper(
                 let login_timeout = attributes.login_timeout.unwrap_or(0);
                 i32_len::set_output_fixed_data(&login_timeout, value_ptr, string_length_ptr)
             }
+            // according to the spec, SQL_ATTR_CONNECTION_DEAD just returns the latest status of the connection, not the current status
+            ConnectionAttribute::SQL_ATTR_CONNECTION_DEAD => {
+                let connection_dead = match *conn.mongo_connection.write().unwrap() {
+                    Some(_) => SqlBool::SQL_FALSE,
+                    None => SqlBool::SQL_TRUE,
+                };
```
Comment (EthanHardyMongo) [inline odbc/src/api/functions.rs:1464]:
```suggestion
            ConnectionAttribute::SQL_ATTR_CONNECTION_DEAD => {
                let connection_dead = match *conn.mongo_connection.read().unwrap() {
                    Some(_) => SqlBool::SQL_FALSE,
                    None => SqlBool::SQL_TRUE,
                };
```
I don't think a write is necessary here

**Review by EthanHardyMongo — APPROVED** [review]
LGTM! I just had one, tiny nit.

File: odbc/src/api/functions.rs (hunk 1 above)
Comment (jchemburkar) [reply odbc/src/api/functions.rs]: good catch. I initially was pinging the database with the connection (hence the write) but realized that was more then the spec was asking for

**Pull request merged**